src/client/providers/jediProxy.ts

```diff
@@ -334,250 +381,201 @@ function spawnProcess(dir: string) {
+            }
+            return Promise.reject(ex);
         }
-        let lines = stdout.split(/\r?\n/g).filter(line => line.length > 0);
-        return validatePath(lines[0]);
-    }).catch(() => {
-        return "";
-    });
-}
-vscode.workspace.onDidChangeConfiguration(onConfigChanged);
-onConfigChanged();
-function onConfigChanged() {
-    // We're only interested in changes to the python path
-    if (lastKnownPythonPath === pythonSettings.pythonPath) {
-        return;
+        return executionCmd.deferred.promise;
     }
 
-    lastKnownPythonPath = pythonSettings.pythonPath;
-    let filePaths = [
-        // Sysprefix
-        getPathFromPythonCommand(["-c", "import sys;print(sys.prefix)"]),
-        // exeucutable path
-        getPathFromPythonCommand(["-c", "import sys;print(sys.executable)"]),
-        // Python specific site packages
-        getPathFromPythonCommand(["-c", "from distutils.sysconfig import get_python_lib; print(get_python_lib())"]),
-        // Python global site packages, as a fallback in case user hasn't installed them in custom environment
-        getPathFromPythonCommand(["-m", "site", "--user-site"]),
-    ];
-
-    let PYTHONPATH: string = process.env['PYTHONPATH'];
-    if (typeof PYTHONPATH !== 'string') {
-        PYTHONPATH = '';
-    }
-    let customEnvironmentVars = getCustomEnvVars();
-    if (customEnvironmentVars && customEnvironmentVars['PYTHONPATH']) {
-        let PYTHONPATHFromEnvFile = customEnvironmentVars['PYTHONPATH'] as string;
-        if (!path.isAbsolute(PYTHONPATHFromEnvFile) && typeof vscode.workspace.rootPath === 'string') {
-            PYTHONPATHFromEnvFile = path.resolve(vscode.workspace.rootPath, PYTHONPATHFromEnvFile);
+    private createPayload<T extends ICommandResult>(cmd: IExecutionCommand<T>): any {
+        var payload = {
+            id: cmd.id,
+            prefix: "",
+            lookup: commandNames.get(cmd.command),
+            path: cmd.fileName,
+            source: cmd.source,
+            line: cmd.lineIndex,
+            column: cmd.columnIndex,
+            config: this.getConfig()
+        };
+
+        if (cmd.command === CommandType.Symbols) {
+            delete payload.column;
+            delete payload.line;
         }
-        PYTHONPATH += (PYTHONPATH.length > 0 ? + path.delimiter : '') + PYTHONPATHFromEnvFile;
+
+        return payload;
     }
-    if (typeof PYTHONPATH === 'string' && PYTHONPATH.length > 0) {
-        filePaths.push(Promise.resolve(PYTHONPATH.trim()));
+
+    private lastKnownPythonPath: string = null;
+    private additionalAutoCopletePaths: string[] = [];
+    private getPathFromPythonCommand(args: string[]): Promise<string> {
+        return execPythonFile(this.pythonSettings.pythonPath, args, this.workspacePath).then(stdout => {
+            if (stdout.length === 0) {
+                return "";
+            }
+            let lines = stdout.split(/\r?\n/g).filter(line => line.length > 0);
```

Comment:
Is there an npm package to do appropriate line splitting? (To start using in the future.)

Reply:
There is one, but not popular enough. I guess the problem is this is a very simple piece of code. Can create a function for this though, but not sure its worth it.

Reply:
Nah, it's fine. I just always assume anything that has been written in JavaScript more than once has a package on npm. 😉